### ISSUE TITLE
const correctness

### ISSUE THREAD

**Issue by superbigio** [x42/libltc#46]
Would you be interested in making libltc const-correct?
I think it's a pretty simple API so it shouldn't be too difficult to do.
I think about a dozen of functions would be affected.

I may have a PR ready for this.
Let me know...

**Comment by x42** [comment]
That'll likely break API and ABI, won't it?

**Comment by superbigio** [comment]
Yes, adding const as a parameter qualifier will break ABI compatibility, but not API.
The API will still be able to compile and work, however there may be a few compiler warnings being thrown at you. (harmless but a little annoying) Although this also depends upon your compiler settings. If you give me permission to send you a PR I will do my best to get rid of all kinds of warnings and annoyances that such a change may produce.
However the ABI will break regardless.

**Comment by lp35** [comment]
Also interested by that
